core/templates/dev/head/editor/ExplorationPreview.js

```diff
@@ -59,5 +61,31 @@ oppia.controller('ExplorationPreview', [
     $scope.$on('playerStateChange', function() {
       $scope.allParams = LearnerParamsService.getAllParams();
     });
+
+    explorationData.getData().then($scope._refreshPreviewTab = function() {
+      var newStateName = explorationInitStateNameService;
+
+      if (angular.equals({}, explorationParamSpecsService.savedMemento)) {
+        newStateName.init(editorContextService.getActiveStateName());
```

Comment:
No -- don't do this; you don't want to change the actual initial state name of the exploration (which this PR is currently doing). Instead, you just want to pass a different value for init_state_name into populateExploration().